DebugServer/Log.h

```diff
@@ -1,17 +1,81 @@
 // SketchUp Ruby API Debugger. Copyright 2013 Trimble Navigation Ltd.
 // Authors:
 // - Bugra Barin
+// - Aaron Hill (@Seraku) - Enhanced logging function.
 //
 #pragma once
 
+#include <chrono>
+#include <iomanip>
+#include <regex>
+#include <sstream>
+#include <string>
+
 namespace SketchUp {
 namespace RubyDebugger {
 
-inline void Log(const char* str) {
+namespace Log {
+
+static inline std::string Context(const char *filename, size_t line, const char *function) {
+  std::ostringstream output;
+
+  static auto start_time = std::chrono::high_resolution_clock::now();
+  auto current_time = std::chrono::high_resolution_clock::now();
+  auto elapsed = std::chrono::duration_cast<std::chrono::microseconds>(current_time - start_time).count() * 0.000001;
```

Comment:
We internally follow Google's C++ style guide but we haven't adhered to that very strictly in this project (It's just been a pet project for a few of us). But we generally wrap lines at 80 characters. Not a big deal for this code but we might want to do a pass later to fix the coding style.

https://google.github.io/styleguide/cppguide.html